src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj

```diff
@@ -19,6 +19,12 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' "/>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' "/>
+  <PropertyGroup>
+    <TargetsBSD Condition="'$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">true</TargetsBSD>
+  </PropertyGroup>
```

Comment:
> Starting another "is BSD" condition into the code makes it very very complicated

I don't see how that's the case.  Yes, it adds another level; I personally think that helps to simplify things, not add complication.  If there's a logical way of saying that a bunch of things are relevant together, then in my opinion that organization helps to lend clarity.  It becomes confusing if we have a bunch of functionality under Unix that doesn't apply to Linux; the case we saw before wasn't about a function not applying to OSX, but rather about that function needing to be specialized for OS X slightly.  In contrast, getmntinfo doesn't exist on Linux at all.

> that will cause more problems than it solves in the future.

If it causes problems in the future, we can revise it in the future. :smile:

> We should stick to the solution we used before.

That made sense when there was no logical grouping that allowed for it to be separated out.  That's not the case here.

> Also OS X is not really a BSD

No one's claiming it's BSD; the claim is that these functions are from BSD.  Both the FreeBSD man page (http://www.freebsd.org/cgi/man.cgi?query=getmntinfo&apropos=0&sektion=0&manpath=FreeBSD+10.1-RELEASE&arch=default&format=html) and the OS X man page (https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/getmntinfo.3.html) agree on that, stating that these functions "function first appeared in 4.4BSD".  And at least for our needs (if not entirely), the function behaves the same way on both platforms, as specified for BSD.

I'm not trying to be difficult, but I'm honestly not understanding the problem.